manager/state/raft/raft.go

```diff
@@ -696,6 +698,12 @@ func (n *Node) ProcessRaftMessage(ctx context.Context, msg *api.ProcessRaftMessa
 		return nil, ErrNoRaftMember
 	}
 
+	// Reject the message if remote member is using a different
+	// address than the one used at join time.
+	if !n.cluster.MemberIsValid(msg.Message.From, msg.Addr) {
```

Comment:
I think it makes more sense to look at the actual address on the GRPC connection (`nodeInfo.RemoteAddr` is added by https://github.com/docker/swarmkit/pull/1121) than rely on self-reported addresses, especially in light of the problem noted in https://github.com/docker/swarmkit/pull/1142#r70011581.